app/models/manageiq/providers/ibm_power_hmc/infra_manager.rb

```diff
@@ -19,7 +19,9 @@ class ManageIQ::Providers::IbmPowerHmc::InfraManager < ManageIQ::Providers::Infr
   supports :metrics
   supports :provisioning
 
-  has_many :hosts_advanced_settings, :through => :hosts, :source => :advanced_settings
+  has_many :hosts_advanced_settings,           :through => :hosts, :source => :advanced_settings
+  has_many :miq_scsi_targets, -> { distinct }, :through => :guest_devices
+  has_many :miq_scsi_luns,    -> { distinct }, :through => :miq_scsi_targets
```

Comment:
The :miq_scsi_luns association can probably be removed as it is identical to the one from the base class.